src/bit-fields/src/bit.rs

```diff
@@ -0,0 +1,817 @@
+                ```
+            ")]
+            #[allow(clippy::integer_arithmetic)]
+            #[inline]
+            pub fn off(&mut self) -> Option<()> {
+                *self.0 = <$y>::new(self.0.get() & !BitMut::<$x, P>::MASK)?;
+                Some(())
+            }
+
+            #[doc = concat!("
+                Returns if the bit is 0.
+                ```
+                use bit_fields::BitMut;
+                let mut x = std::num::", stringify!($y), "::new(5).unwrap();
+                assert_eq!(BitMut::<_,0>(&mut x).is_off(),false);
+                assert_eq!(BitMut::<_,1>(&mut x).is_off(),true);
+                assert_eq!(BitMut::<_,2>(&mut x).is_off(),false);
+                ```
+            ")]
+            #[must_use]
+            #[inline]
+            pub fn is_off(&self) -> bool {
+                !bool::from(self)
+            }
+
+            // These values are only evaluated at compile-time, thus a failure can only occur at
+            // compile-time and would be immediately obvious. Thus it is safe to use arithmetic
+            // here.
+            #[doc = concat!("
+                Flips the value of the bit.
+                ```
+                use bit_fields::BitMut;
+                use std::num::", stringify!($y), ";
+                let mut x = ", stringify!($y), "::new(5).unwrap();
+                assert_eq!(BitMut::<_,0>(&mut x).flip(),Some(()));
+                assert_eq!(x,", stringify!($y), "::new(4).unwrap());
+                assert_eq!(BitMut::<_,1>(&mut x).flip(),Some(()));
+                assert_eq!(x,", stringify!($y), "::new(6).unwrap());
+                assert_eq!(BitMut::<_,2>(&mut x).flip(),Some(()));
+                assert_eq!(x,", stringify!($y), "::new(2).unwrap());
+                ```
+            ")]
+            #[allow(clippy::integer_arithmetic)]
+            #[inline]
+            pub fn flip(&mut self) -> Option<()> {
+                *self.0 = <$y>::new(self.0.get() ^ BitMut::<$x, P>::MASK)?;
+                Some(())
+            }
+
+            #[doc = concat!("
+                Sets the bit.
+                ```
+                use bit_fields::BitMut;
+                use std::num::", stringify!($y), ";
+                let mut x = ", stringify!($y), "::new(5).unwrap();
+                BitMut::<_,0>(&mut x).set(false);
+                assert_eq!(x,", stringify!($y), "::new(4).unwrap());
+                BitMut::<_,1>(&mut x).set(true);
+                assert_eq!(x,", stringify!($y), "::new(6).unwrap());
+                BitMut::<_,2>(&mut x).set(false);
+                assert_eq!(x,", stringify!($y), "::new(2).unwrap());
+                ```
+            ")]
+            #[inline]
+            pub fn set(&mut self, x: bool) -> Option<()> {
+                if x {
+                    self.on();
+                    Some(())
+                } else {
+                    self.off()
+                }
```

Comment:
Same here:
```rust
*self.0 = <$y>::new((self.0.get() & !BitMut::<$x, P>::MASK) | (x as $x) << P)?;
```

Reply:
I think this is a similar discussion to https://github.com/firecracker-microvm/firecracker/pull/3105#discussion_r1050624357